src/upgrade.c

```diff
@@ -22,6 +22,8 @@ struct upgrade_ctx {
 	unsigned short solver_flags;
 	int no_self_upgrade : 1;
 	int self_upgrade_only : 1;
+  	unsigned int ignore;
+  	const char* i_pkg[10];
```

Comment:
I'm wondering if using a limited size array with an arbitrary limit is a good choice. 
My thought is that if a user wants to ignore more than X packages (here 10), it's more interesting to manually upgrade the other packages.
I also think that using a dynamic structure would be a little bit overkill for the use case.
Any advice on this ?